catalog/internal/catalog/performance_metrics.go

```diff
@@ -96,3 +122,362 @@ func LoadPerformanceMetricsData(path []string, modelRepo dbmodels.CatalogModelRe
+					Name:        key,
+					StringValue: &jsonStr,
+				})
+			}
+		}
+	}
+
+	// Add metadata file path for reference
+	metadataFilePath := filepath.Join(dirPath, "metadata.json")
+	customProperties = append(customProperties, models.Properties{
+		Name:        "metadata_file_path",
+		StringValue: &metadataFilePath,
+	})
+
+	// Create the model with the provided TypeID from the type map
+	model := &dbmodels.CatalogModelImpl{
+		ID:     existingID, // Use existing ID if updating
+		TypeID: &typeID,
+		Attributes: &dbmodels.CatalogModelAttributes{
+			Name:                     &metadata.ID,
+			ExternalID:               &metadata.ID,
+			CreateTimeSinceEpoch:     createTime,
+			LastUpdateTimeSinceEpoch: updateTime,
+		},
+		Properties:       &properties,
+		CustomProperties: &customProperties,
+	}
+
+	return model
+}
+
+// createAPIModelFromMetadata converts metadata to API model for backward compatibility
+func createAPIModelFromMetadata(metadata metadataJSON, dirPath string) *model.CatalogModel {
+	// Convert timestamps to string format
+	var createTime, updateTime *string
+	if metadata.CreatedAt > 0 {
+		createTimeStr := strconv.FormatInt(metadata.CreatedAt, 10)
+		createTime = &createTimeStr
+	}
+	if metadata.UpdatedAt > 0 {
+		updateTimeStr := strconv.FormatInt(metadata.UpdatedAt, 10)
+		updateTime = &updateTimeStr
+	}
+
+	// Create custom properties from the dynamically captured fields
+	customProps := make(map[string]model.MetadataValue)
+
+	// Add all custom properties from the CustomProperties map
+	for key, value := range metadata.CustomProperties {
+		// Handle different value types
+		switch v := value.(type) {
+		case string:
+			if v != "" {
+				customProps[key] = model.MetadataStringValueAsMetadataValue(
+					model.NewMetadataStringValue(v, "MetadataStringValue"),
+				)
+			}
+		case []interface{}, map[string]interface{}:
+			// For arrays and objects, marshal to JSON string
+			jsonBytes, err := json.Marshal(v)
+			if err == nil {
+				customProps[key] = model.MetadataStringValueAsMetadataValue(
+					model.NewMetadataStringValue(string(jsonBytes), "MetadataStringValue"),
+				)
+			}
+		case float64:
+			// Numbers come as float64 from JSON
+			customProps[key] = model.MetadataDoubleValueAsMetadataValue(
+				model.NewMetadataDoubleValue(v, "MetadataDoubleValue"),
+			)
+		case int64:
+			intStr := strconv.FormatInt(v, 10)
+			customProps[key] = model.MetadataIntValueAsMetadataValue(
+				model.NewMetadataIntValue(intStr, "MetadataIntValue"),
+			)
+		case bool:
+			customProps[key] = model.MetadataBoolValueAsMetadataValue(
+				model.NewMetadataBoolValue(v, "MetadataBoolValue"),
+			)
+		default:
+			// For any other type, try to marshal to JSON string
+			jsonBytes, err := json.Marshal(v)
+			if err == nil {
+				customProps[key] = model.MetadataStringValueAsMetadataValue(
+					model.NewMetadataStringValue(string(jsonBytes), "MetadataStringValue"),
+				)
+			}
+		}
+	}
```

Comment:
It might be worth converting this mapping into a utility function which can take a single value and use the `reflect` package to map it to the correct metadata type